js/lib/AlphabeticIndex.js

```diff
@@ -60,159 +62,322 @@ var Collator = require("./Collator.js");
-		}
-		
-		if (typeof(options.sync) !== 'undefined') {
-			this.sync = (options.sync == true);
-		}
-		
-		if (options.loadParams) {
-			this.loadParams = options.loadParams;
-		}
-	}
-
-	this.locale = this.locale || new Locale();
-
-	//console.log("implemented in pure JS");
-	if (!Collator.cache) {
-		Collator.cache = {};
-	}
-
-	Utils.loadData({
-		object: Collator, 
-		locale: this.locale, 
-		name: "collation.json",
-		sync: sync,
-		loadParams: loadParams, 
-		callback: ilib.bind(this, function (collation) {
-			if (!collation) {
-				collation = ilib.data.collation;
-				var spec = this.locale.getSpec().replace(/-/g, '_');
-				Collator.cache[spec] = collation;
-			}
-			this._init(collation);
-			new LocaleInfo(this.locale, {
-				sync: sync,
-				loadParams: loadParams,
-				onLoad: ilib.bind(this, function(li) {
-					this.li = li;
-					if (this.ignorePunctuation) {
-		    			isPunct._init(sync, loadParams, ilib.bind(this, function() {
-							if (options && typeof(options.onLoad) === 'function') {
-								options.onLoad(this);
-							}
-		    			}));
-	    			} else {
-						if (options && typeof(options.onLoad) === 'function') {
-							options.onLoad(this);
-						}
-	    			}
-	    		})
-			});
-		})
-	});
+    this.sync = true;
+    this.loadParams = {};
+    this.caseSensitive = false;
+    this.accentSensitive = false;
+    this.overflowLabel = "#";
+    this.underflowLabel = "*";
+    this.style = "standard";    
+    this.index = [];
+
+    if (options) {
+        if (options.locale) {
+            this.locale = (typeof(options.locale) === 'string') ? new Locale(options.locale) : options.locale;
+        }
+        
+        if (typeof(options.style) !== 'undefined') {
+            this.style = options.style;
+        }
+
+        if (typeof(options.overflowLabel) !== 'undefined') {
+            this.overflowLabel = options.overflowLabel;
+        }
+
+        if (typeof(options.underflowLabel) !== 'undefined') {
+            this.underflowLabel = options.underflowLabel;
+        }
+        
+        if (typeof(options.sync) !== 'undefined') {
+            this.sync = (options.sync == true);
+        }
+        if (options.loadParams) {
+            this.loadParams = options.loadParams;
+        }
+    }
+
+    this.locale = this.locale || new Locale();
+    
+    isAscii._init(this.sync, this.loadParams, ilib.bind(this, function() {
+        isIdeo._init(this.sync, this.loadParams, ilib.bind(this, function(){
+            isDigit._init(this.sync, this.loadParams, ilib.bind(this, function(){
+                new Collator ({
+                    locale: this.locale,
+                    useNative: false,
+                    sensivitiy: "primary",
+                    usage: "sort",
+                    sync: this.sync,
+                    loadParam : this.loadParams,
+                    onLoad: ilib.bind(this, function(collation){
+                        this.collationObj = collation;
```

Comment:
Should we call NormString.init here to load the files needed? If so, NormString.init should be fixed so that it only initializes the first time it is called. The second time, it should not do all the work over again.

Reply:
I just want to make sure it means changing the implementation of NormString.init() same as CType._load(), Did I get it exactly?

Reply:
Yes, exactly!

Reply:
oh, I see. so I'll re-update codes.